### DISCUSSION TITLE
Timeout

### DISCUSSION THREAD

**Discussion by djpitucco** [viniciussanchez/RESTRequest4Delphi#164] [General]
Boa tarde Vinicius, saberia me dizer se consigo amentar o timeout de aguardo de resposta? Pelo que vi por padrão ele espera uns 20 segundos.

**Comment by viniciussanchez** [comment]
só colocar um .Timeout()

**Comment by djpitucco** [comment]
Estaria certo dessa forma ou ele tem algum lugar especifico que deva estar? Pelo teste que fiz ele ficou os mesmos 20 segundos.

    LResponse := TRequest.New.BaseURL('http://'+ip+'/export_afd.fcgi?session='+session)
    .Accept('application/json')
    .ClearBody
    .AddBody(ComandoJSON)
    .Timeout(60000) //vai esperar a resposta 60 segundos
    .post ;

  if LResponse.StatusCode = 200 then
  begin

     retorno := '';
     retorno := LResponse.Content;
end;

**Reply by viniciussanchez** [reply]
Aparentemente está correto

**Comment by djpitucco** [comment]
Grato pelo retorno!
